src/tests/Microsoft.PowerFx.Core.Tests.Shared/ExpressionTestCases/AsType_UO.txt

```diff
@@ -0,0 +1,94 @@
+true
+
+>> AsType(ParseJSON("""AstypeFunction"""), Type(Text))
+"AstypeFunction"
+
+>> AsType(ParseJSON("""1984-01-01"""), Date)
+Date(1984,1,1)
+
+>> AsType(ParseJSON("""1900-12-31T23:59:59.999Z"""), DateTime)
+DateTime(1900,12,31,23,59,59,999)
+
+>> If(AsType(ParseJSON("false"), Boolean), "MyFalse", "MyTrue")
+"MyTrue"
+
+>> AsType(ParseJSON("42"), Number) = 42
+true
+
+>> UniChar(AsType(ParseJSON("66"), Number))
+"B"
+
+// record
+>> AsType(ParseJSON("{""foo"": true, ""bar"": 1.1}"), Type({foo: Boolean, bar: Number}))
+{bar:1.1,foo:true}
+
+// record missing field
+>> AsType(ParseJSON("{""Name"": ""SpongeBob"", ""Age"": 1}"), Type({Name: Text, Age: Number, Aquatic: Boolean})).Name
+"SpongeBob"
+
+// Deeply nested record with table
+>> AsType(ParseJSON("{""a"": {""b"" : { ""c"" : [1, 2, 3, 4]}}}"), Type({a: {b: {c: [Number]}}}))
+{a:{b:{c:Table({Value:1},{Value:2},{Value:3},{Value:4})}}}
+
+// Table
+>> AsType(ParseJSON("[{""a"": ""Hello"", ""b"": ""2012-01-02""}, {""a"": ""Hi"", ""b"": ""2012-01-03""}]"), Type([{a: Text, b: Date}]))
+Table({a:"Hello",b:Date(2012,1,2)},{a:"Hi",b:Date(2012,1,3)})
+
+>> AsType(ParseJSON("[{""a"": [{""z"": true}, {""z"": false}]}, {""a"": [{""z"": false}, {""z"": true}]}]"), Type([{a: [{z: Boolean}]}]))
+Table({a:Table({z:true},{z:false})},{a:Table({z:false},{z:true})})
+
+// Negative tests
+>> AsType(ParseJSON("5"), Text)
+Error({Kind:ErrorKind.InvalidJSON})
+
+>> AsType(ParseJSON("""42"""), Number)
+Error({Kind:ErrorKind.InvalidJSON})
+
+>> AsType(ParseJSON("""1900-12-31T24:59:59.1002Z"""), DateTime)
+Error({Kind:ErrorKind.InvalidJSON})
+
+>> AsType(ParseJSON("1"), 1)
+Errors: Error 23-24: Invalid argument '1'. Expected valid type name or type literal.
+
+>> AsType(ParseJSON("5"), Type(5))
+Errors: Error 28-29: Type literal declaration is invalid. The expression '5' cannot be used in a type definition.|Error 27-28: Type literal declaration is invalid. The expression 'Type(5)' cannot be used in a type definition.
+
+>> AsType(ParseJSON("true"), UnKnown)
+Errors: Error 26-33: Name isn't valid. 'UnKnown' isn't recognized.|Error 0-34: Invalid argument 'UnKnown'. Expected valid type name or type literal.
+
+>> AsType(ParseJSON("fasle"), Boolean)
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> AsType(ParseJSON("true"), Void)
+Errors: Error 26-30: Invalid argument type (Void). Expecting a Error value instead.|Error 0-6: The function 'AsType' has some invalid arguments.
+
+>> AsType(ParseJSON("{""a"": 5, ""b"":true}"), Type({a: Number}))
+Error({Kind:ErrorKind.InvalidJSON})
+
+>> AsType(ParseJSON("{""a"": 5}"), Type({a: Number}), "Hello")
+Errors: Error 0-59: Invalid number of arguments: received 3, expected 2.
+
+>> AsType(ParseJSON("true"), None)
+Errors: Error 26-30: Unsupported untyped/JSON conversion type 'ObjNull' in argument.
+
+>> AsType(ParseJSON("{}"), Type({a: Text, b: [Color]}))
+Errors: Error 28-29: Unsupported untyped/JSON conversion type 'Color' in argument.
```

Comment:
>Unsupported untyped/JSON conversion type 'Color' in argument. [](http://example.com/codeflow?start=21&length=61)

Is this a problem with the Color type (i.e., if the second argument were `Type({a:Text,b:Number})`, would it work)? If so, then we should have a negative test with Color as a primitive too. If not, then why wouldn't this work? From line 33, we can have missing fields in the record.